### ISSUE TITLE
Heavy memory usage

### ISSUE THREAD

**Issue by alexvonsiebenthal** [supermerill/SuperSlicer#2774]
### What happened?

Superslicer eats lots of RAM on my M1 (Pro) MacBook Pro. 2 instances are running, one of which is empty. The empty one consumes over 2GB of memory, the one in used with some models loaded over 4GB.

Is this known behaviour? What eats all that memory? Can this be optimised?
<img width="950" alt="Screenshot 2022-05-01 at 22 46 52" src="https://user-images.githubusercontent.com/6860125/166164076-8e66e3ad-4f00-45a1-b934-7e66b1b3b68a.png">

### Project file & How to reproduce

just open and use SuperSlicer 🤷 

### Version

2.4.58.3

### Operating system

macOS 12.3

### Printer model

voron 2.4 / stealthburner

**Comment by supermerill** [comment]
Is it a problem?

fusion also use many memory.

**Comment by alexvonsiebenthal** [comment]
Yes, but fusion supposedly has to do lots more than superslicer :-) it is a problem in terms of overall resource management. I even had SuperSlicer instances crash (I think terminated by the OS for going over board with memory usage)

**Comment by supermerill** [comment]
superslicer has to store each facet generated by fusion360, multiplied by the number of layers, multiplied by the number of perimeters, and also the paths for all infill moves.

I advise to use a `resolution` setting to avoid using models with too many details.

**Comment by alexvonsiebenthal** [comment]
> superslicer has to store each facet generated by fusion360, multiplied by the number of layers, multiplied by the number of perimeters, and also the paths for all infill moves.
> 
> I advise to use a `resolution` setting to avoid using models with too many details.

Fair enough. However, it doesn't explain why an empty instance consumes over 2GB of RAM.

**Comment by supermerill** [comment]
> Fair enough. However, it doesn't explain why an empty instance consumes over 2GB of RAM.

On my windows & linux, it's some mio.
I blame macos.
Maybe you're using a intel version on arm?

**Comment by alexvonsiebenthal** [comment]
It is the intel version indeed (nightly). Might have fetched the wrong one then and will check with the ARM version.

**Comment by supermerill** [comment]
the arm currently isn't buildable by servers. You have to compile it manually.
Here is the latest I built: https://mega.nz/file/NxIigRwR#hKgPdD1LykfLO3cwrUt_Qzs4BvcDVgh94_KDSNcV788

**Comment by blalor** [comment]
As a point of comparison, PrusaSlicer 2.4.2+arm64 (running on my M1 MacBook Air) is using < 1G with a model loaded and sliced, and it's been open and idle for a day. 
<img width="737" alt="image" src="https://user-images.githubusercontent.com/109915/168052244-da2659d8-60e4-4eb0-ba1a-b7164741f80d.png">

**Comment by supermerill** [comment]
<img width="1258" alt="Screenshot 2022-08-11 at 03 38 34" src="https://user-images.githubusercontent.com/6536403/183991248-51d987a7-ce11-46ae-8af3-68e4587f39e1.png">
For the record, my new dev arm version on arm Mac. Should already be an improvement.
